Automattic/woocommerce-payments
# fix: ensure BNPL enablement is not adding unnecessary currencies

#### Changes proposed in this Pull Request









When BNPL methods are enabled (either via onboarding or via the settings pages), we're adding unnecessary currencies.
BNPL methods are set only to accept "domestic payments". I.e.: customers can't use BNPL unless they're paying with the same currency as the merchant's account currency.

Since BNPL methods are configured to only accept "domestic" payments, we should only add the "domestic" currency if needed.

This PR fixes the backend logic and the UI when removing currencies from the multi-currency settings.

Before - Klarna shouldn't be disabled if EUR is disabled and the account's default currency is USD:
![Screenshot 2024-02-15 at 3 42 33 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/4b114413-85ad-44ef-b060-e28dfc465adc)

After - Klarna is no longer displayed in the modal:
![Screenshot 2024-02-15 at 3 41 24 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/29e6bb09-b256-4ae2-a35c-a9475ad7c31c)

The messaging on onboarding is fixed here: https://github.com/Automattic/woocommerce-payments/pull/8202
The messaging on the settings page when the payment method requires additional setup will be fixed on a separate PR.

#### Testing instructions





### Test 1
- Ensure you have Klarna, Afterpay/Clearpay, Affirm disabled. Ensure you only have CC enabled.
- Take note of your Stripe account currency - let's say it's USD
- Make sure you have only USD (as a default currency) in the multi-currency settings
- Go to http://wcpay.test/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods
- Enable any of the BNPL methods, click "continue" (ignore the messaging in the UI, if https://github.com/Automattic/woocommerce-payments/pull/8202 hasn't been merged yet)
- Go to the multi-currency settings
- Ensure USD is the only enabled currency

### Test 2
- Ensure you have Klarna, Afterpay/Clearpay, Affirm disabled. Ensure you only have CC enabled.
- Take note of your Stripe account currency - let's say it's USD
- Change the default WC currency to "EUR" & ensure you have EUR (as a default currency) in the multi-currency settings
- Go to http://wcpay.test/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fadditional-payment-methods
- Enable any of the BNPL methods & giropay, click "continue"
- Go to the multi-currency settings
- Ensure you have EUR & USD enabled
- Change the store's default currency to USD, to allow removing EUR in the settings
- Go to the multi-currency settings
- Attempt to remove "EUR"
- The modal should only show `giropay` as a PM, and not Klarna
![Screenshot 2024-02-15 at 3 41 24 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/29e6bb09-b256-4ae2-a35c-a9475ad7c31c)


-------------------


**Post merge**
